### PR TITLE
Generalize error pages and add response padding

### PR DESCRIPTION
**Pull request by arothuis**
#### Related PR
[OpenConext-deploy PR #89](https://github.com/OpenConext/OpenConext-deploy/pull/89)

#### Issues
[127988865](https://www.pivotaltracker.com/story/show/127988865)
[127989511](https://www.pivotaltracker.com/story/show/127989511)
[127989925](https://www.pivotaltracker.com/story/show/127989925)

### CODE REVIEW

File: src/OpenConext/EngineBlockBundle/EventListener/ErrorResponseTimePaddingListener.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                'Caught Exception "%s":"%s", not padding response time: '
+                . 'it exceeds the configured padded response time (%d seconds) by %f seconds',
+                get_class($exception),
+                $exception->getMessage(),
+                $this->paddedResponseTime,
+                $responseTimePaddingInSeconds
+            ));
+        }
+
+        $message         = 'Unable to verify message';
```
Comment (arothuis) [inline src/OpenConext/EngineBlockBundle/EventListener/ErrorResponseTimePaddingListener.php]: Alternatively, we could return here and let the RedirectToFeedbackPageExceptionListener take care of the redirecting. This way, this Listener would only be responsible for padding certain marked exceptions.

File: src/OpenConext/EngineBlockBundle/EventListener/ErrorResponseTimePaddingListener.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        }
+
+        $this->logger->warning(
+            sprintf('Handling exception: "%s": "%s"', get_class($exception), $exception->getMessage())
+        );
+
+        if ($this->executionTimeTracker->currentExecutionTimeExceeds($this->minimumExecutionTime)) {
+            $this->logger->warning(sprintf(
+                'Not padding response time: it exceeds the configured padded response time (%d seconds)',
+                $this->minimumExecutionTime
```
Comment (DRvanR) [inline src/OpenConext/EngineBlockBundle/EventListener/ErrorResponseTimePaddingListener.php]: This argument is not correct, string contains `(%d seconds)`, this is an object.

**Comment by DRvanR** [comment]
One minor thing, the rest is looking good! Please hold off on merging till the 5.0.0 release has been tagged (will ping when that happened) so that we can keep a clean release :smile:

**Comment by DRvanR** [comment]
(and please reference the accompanying PR on Deploy :wink: )

File: src/OpenConext/EngineBlockBundle/EventListener/ErrorResponseTimePaddingListener.php (hunk 2 above)
Comment (arothuis) [inline src/OpenConext/EngineBlockBundle/EventListener/ErrorResponseTimePaddingListener.php]: Well spotted! Forgot to change it when the value object got introduced.

**Comment by arothuis** [comment]
@DRvanR Good to go?

File: src/OpenConext/EngineBlockBundle/EventListener/ExecutionTimeTracker.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace OpenConext\EngineBlockBundle\EventListener;
+
+use OpenConext\EngineBlockBundle\Value\ExecutionTime;
+use Symfony\Component\Stopwatch\Stopwatch;
+
+final class ExecutionTimeTracker
+{
+    const NAME = 'execution-time';
```
Comment (DRvanR) [inline src/OpenConext/EngineBlockBundle/EventListener/ExecutionTimeTracker.php]: `SECTION-NAME`?

**Comment by DRvanR** [comment]
two nits, other than that :+1:

File: src/OpenConext/EngineBlockBundle/EventListener/ErrorResponseTimePaddingListener.php [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+use OpenConext\EngineBlockBridge\ErrorReporter;
+use OpenConext\EngineBlockBundle\Exception\AddExecutionTimePadding;
+use OpenConext\EngineBlockBundle\Value\ExecutionTime;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+class ErrorResponseTimePaddingListener
```
Comment (arothuis) [inline src/OpenConext/EngineBlockBundle/EventListener/ErrorResponseTimePaddingListener.php]: Will make it final.
Will also rename it to something more related to the ExecutionTime: `ExecutionTimePaddingListener`, perhaps.

**Pull request merged**